misc/python/materialize/cli/scratch/destroy.py

```diff
@@ -8,26 +8,73 @@
 # by the Apache License, Version 2.0.
 
 import argparse
+import sys
+from typing import List
 
 import boto3
+from mypy_boto3_ec2.type_defs import FilterTypeDef
 
 from materialize.cli.scratch import check_required_vars
-from materialize.scratch import print_instances
+from materialize.scratch import print_instances, whoami
 
 
 def configure_parser(parser: argparse.ArgumentParser) -> None:
     parser.add_argument(
         "instances",
-        nargs="+",
+        nargs="*",
         help="Instance IDs to destroy",
     )
+    parser.add_argument(
+        "--all-mine",
+        action="store_true",
+        help="Destroy all of your instances (incompatible with specifying instance IDs)",
+    )
+    parser.add_argument(
+        "-n",
+        "--dry-run",
```

Comment:
in destructive cases like this, I actually prefer a `--no-dry-run` flag, and defaulting to the dry-run, thoughts?

Reply:
Ack, yeah, good idea. In discussion with @umanwizard I've changed it so the default behavior is to ask for confirmation, but you can pass `-y` or `--yes` to skip the explicit confirmation.